crates/ruff/src/rules/pyflakes/fixes.rs

```diff
@@ -18,34 +23,20 @@ pub fn remove_unused_format_arguments_from_dict(
     stylist: &Stylist,
 ) -> Result<Fix> {
     let module_text = locator.slice(stmt);
-    let mut tree = match_module(module_text)?;
-    let mut body = match_expr(&mut tree)?;
-
-    let new_dict = {
-        let Expression::Dict(dict) = &body.value else {
-            bail!("Expected Expression::Dict")
-        };
-
-        Dict {
-            lbrace: dict.lbrace.clone(),
-            lpar: dict.lpar.clone(),
-            rbrace: dict.rbrace.clone(),
-            rpar: dict.rpar.clone(),
-            elements: dict
-                .elements
-                .iter()
-                .filter_map(|e| match e {
-                    DictElement::Simple {
-                        key: Expression::SimpleString(name),
-                        ..
-                    } if unused_arguments.contains(&raw_contents(name.value)) => None,
-                    e => Some(e.clone()),
-                })
-                .collect(),
-        }
-    };
-
-    body.value = Expression::Dict(Box::new(new_dict));
+    let mut tree = match_expression(module_text)?;
+    let dict = match_dict(&mut tree)?;
+
+    dict.elements = dict
+        .elements
+        .iter()
+        .filter_map(|e| match e {
+            DictElement::Simple {
+                key: Expression::SimpleString(name),
+                ..
+            } if unused_arguments.contains(&raw_contents(name.value)) => None,
+            e => Some(e.clone()),
+        })
+        .collect();
```

Comment:
We should be able to avoid cloning by using `std::mem::take` and `into_iter`

```suggestion
	let current_elements = std::mem::take(&mut dict.elements);
    dict.elements = current_elements
        .into_iter()
        .filter_map(|e| match e {
            DictElement::Simple {
                key: Expression::SimpleString(name),
                ..
            } if unused_arguments.contains(&raw_contents(name.value)) => None,
            e => Some(e),
        })
        .collect();
```

Or you may be able to use `dict.elements.retain(|e| ...)` instead

Reply:
Done.